packages/openapi3-parser/lib/parser.js

```diff
@@ -126,14 +128,35 @@ function parse(source, context) {
     R.unless(isObjectOrAnnotation, createError(context.namespace, 'Source document is not an object')),
     R.unless(isAnnotation, parseOpenAPIObject(context)));
 
-  return R.chain(
+  const parseResult = R.chain(
     R.pipe(
       parseDocument,
       deduplicateUnsupportedAnnotations(context.namespace),
       context.options.generateSourceMap ? filterColumnLine : filterSourceMaps
     ),
     document
   );
+
+  if (!isAnnotation(parseResult.content[0])) {
```

Comment:
This works a bit different than in API Blueprint version,
if I remove it it will cause 2 tests to fail.

```
  1) #parse
       fails to parse an OAS3 document with invalid YAML:
     TypeError: Cannot read property 'find' of undefined

  2) #parse
       fails to parse a non-object YAML document:
     TypeError: Cannot read property 'find' of undefined
```

This because in case of invalid Yaml there is no version of OAS3 to be parsed.
On the other hand, if there is a parse result containing errors, it will still be able to read the format version contained in the API Category (`parseResult.content[0]`) as Annotation will be (`parseResult.content[1]`).
How do you suggest to resolve this?

Reply:
removed